src/metabase/lib/portable.cljc

```diff
@@ -0,0 +1,124 @@
+(ns metabase.lib.portable
+  (:require
+   [medley.core :as m]
+   [metabase.lib.metadata :as lib.metadata]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.util.match :as lib.util.match]
+   [metabase.lib.walk :as lib.walk]
+   [metabase.util.malli :as mu]
+   [metabase.util.malli.registry :as mr]))
+
+(defprotocol ExportVisitor
+  (export-database     [this mp context database-id])
+  (export-source-table [this mp context source-table-id])
+  (export-source-card  [this mp context source-card-id])
+  (export-field        [this mp context field-id]))
```

Comment:
I think all of these names would be better as `export-foo-id`, and likewise for the `ImportVisitor`.

That's more precise (it's a `database-id`, not a `:metadata/database`) and it doesn't over-specify where any particular table or card ID was found. (`:source-table` and `:source-card` are certainly the key places, but they are not the only ones.)

Reply:
What are the other places we might encounter a table or card id?